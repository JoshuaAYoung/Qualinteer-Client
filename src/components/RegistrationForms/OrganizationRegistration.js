import React, { Component } from 'react';
import UserFields from './UserFields'
import ValidationError from '../../utilities/ValidationError';

export default class OrganizationRegistration extends Component {
  render() {
    const { zipcodeError } = this.props.errorProps
    return (
      <div>
        <header role="banner" className="checkpointBanner">
          <h1 className='pageHeader'>Find Help.</h1>
          <p className='pageInstructions'>All fields are required unless noted otherwise.</p>
        </header>
        <UserFields
          addTempValidation={this.props.addTempValidation}
          addTempValue={this.props.addTempValue}
          errorProps={this.props.errorProps}
        />
        <div>
          <label
            htmlFor='name'
            className='registerLabel'>
            Organization Name:
            </label>
          <br />
          <input
            type='text'
            name='name'
            id='name'
            className='registerInput'
            onChange={ev => this.props.addTempValue('name', ev.target.value)}
            required
          />
        </div>
        <br />
        <div>
          <label
            htmlFor='address'
            className='registerLabel'>
            Address (optional):
            </label>
          <br />
          <input
            type='text'
            name='address'
            id='address'
            className='registerInput'
            onChange={ev => this.props.addTempValue('address', ev.target.value)}
          />
        </div>
        <br />
        <div>
          <label
            htmlFor='city'
            className='registerLabel'>
            City:
            </label>
          <br />
          <input
            type='text'
            name='city'
            id='city'
            className='registerInput'
            onChange={ev => this.props.addTempValue('city', ev.target.value)}
            required
          />
        </div>
        <br />
        <div>
          <label
            htmlFor='state'
            className='registerLabel'>
            State:
            </label>
          <br />
          <input
            type='text'
            name='state'
            id='state'
            className='registerInput'
            onChange={ev => this.props.addTempValue('state', ev.target.value)}
            required
          />
        </div>
        <br />
        <div>
          <label
            htmlFor='zipcode'
            className='registerLabel'>
            Zipcode:
          </label>
          <br />
          <input
            type='text'
            name='zipcode'
            id='zipcode'
            className='registerInput'
            onChange={ev => this.props.addTempValidation('zipcode', ev.target.value)}
            required
          />
        </div>
        {zipcodeError && <ValidationError message={zipcodeError} />}
        <br />
        <div>
          <label
            htmlFor='phone'
            className='registerLabel'>
            Phone Number (optional):
            </label>
          <br />
          <input
            type='text'
            name='phone'
            id='phone'
            className='registerInput'
            onChange={ev => this.props.addTempValue('phone', ev.target.value)}
          />
        </div>
        <br />
        <div>
          <label
            htmlFor='website'
            className='registerLabel'>
            Website (optional):
            </label>
          <br />
          <input
            type='text'
            name='website'
            id='website'
            className='registerInput'
            onChange={ev => this.props.addTempValue('website', ev.target.value)}
          />
        </div>
      </div >
    )
  }
}