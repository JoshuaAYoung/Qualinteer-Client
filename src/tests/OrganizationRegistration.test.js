import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OrganizationRegistration from '../components/RegistrationForms/OrganizationRegistration';

describe('OrganizationRegistration Component', () => {
  const props = {
    errorProps: {
      zipcodeError: 'Test Error',
    },
  };

  it('OrganizationRegistration Smoke Test', () => {
    const div = document.createElement('div');

    ReactDOM.render(<OrganizationRegistration {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('OrganizationRegistration Snapshot Test', () => {
    const wrapper = shallow(<OrganizationRegistration {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
