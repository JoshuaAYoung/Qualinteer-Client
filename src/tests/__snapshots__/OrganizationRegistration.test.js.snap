// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationRegistration Component OrganizationRegistration Snapshot Test 1`] = `
<div>
  <UserFields
    errorProps={
      Object {
        "zipcodeError": "Test Error",
      }
    }
  />
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Organization Name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="address"
      name="address"
      onChange={[Function]}
      placeholder="Address"
      type="text"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="city"
      name="city"
      onChange={[Function]}
      placeholder="City"
      required={true}
      type="text"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="state"
      name="state"
      onChange={[Function]}
      placeholder="State"
      required={true}
      type="text"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="zipcode"
      name="zipcode"
      onChange={[Function]}
      placeholder="Zipcode"
      required={true}
      type="text"
    />
    <ValidationError
      message="Test Error"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="phone"
      name="phone"
      onChange={[Function]}
      placeholder="Phone Number (optional)"
      type="text"
    />
  </div>
  <div
    className="inputContainer"
  >
    <input
      className="registrationInput"
      id="website"
      name="website"
      onChange={[Function]}
      placeholder="Website (optional)"
      type="text"
    />
  </div>
</div>
`;
